internal/connector/internal/migrations/202208220000_add_connector_type_lease.go

```diff
@@ -0,0 +1,44 @@
+package migrations
+
+// Migrations should NEVER use types from other packages. Types can change
+// and then migrations run on a _new_ database will fail or behave unexpectedly.
+// Instead of importing types, always re-create the type in the migration, as
+// is done here, even though the same type is defined in pkg/api
+
+import (
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/api"
+	"github.com/bf2fc6cc711aee1a0c2a/kas-fleet-manager/pkg/db"
+	"github.com/go-gormigrate/gormigrate/v2"
+	"gorm.io/gorm"
+	"time"
+)
+
+func addConnectorTypeLease(migrationId string) *gormigrate.Migration {
+
+	type LeaderLease struct {
+		db.Model
+		Leader    string
+		LeaseType string
+		Expires   *time.Time
+	}
+
+	return db.CreateMigrationFromActions(migrationId,
+		db.FuncAction(func(tx *gorm.DB) error {
+			// We don't want to delete the leader lease table on rollback because it's shared with the kas-fleet-manager
+			// so we just create it here if it does not exist yet.. but we don't drop it on rollback.
```

Comment:
It's not true anymore because the DBs are separate. The comment was copied from an older lease migration code.

Reply:
The comment was more on sharing the database with kfm. It okay for that part to be removed

Reply:
> Thanks for the explanation @dhirajsb
> 
> Just to understand more on the deadlock and it how it happened.
> 
> > ..We only want the lease for connector types to expire, since that's the cause of the deadlock
> 
> ...the deadlock was that no any incoming http requests would be served since the connectors types weren't reconciled yet?

That's correct.